gateway-ha/src/test/java/io/trino/gateway/ha/router/TestSpecificDbResourceGroupsManager.java

```diff
@@ -40,30 +40,53 @@ final class TestSpecificDbResourceGroupsManager
     @Override
     void setUp()
     {
-        specificDb = "h2db-" + System.currentTimeMillis();
-        File tempH2DbDir = Path.of(System.getProperty("java.io.tmpdir"), specificDb).toFile();
-        tempH2DbDir.deleteOnExit();
-        String jdbcUrl = "jdbc:h2:" + tempH2DbDir.getAbsolutePath();
-        HaGatewayTestUtils.seedRequiredData(tempH2DbDir.getAbsolutePath());
-        DataStoreConfiguration db = new DataStoreConfiguration(jdbcUrl, "sa",
-                "sa", "org.h2.Driver", 4, false);
-        Jdbi jdbi = Jdbi.create(jdbcUrl, "sa", "sa");
+        specificDb = "test_db_" + System.currentTimeMillis();
+        PostgreSQLContainer<?> postgres = new PostgreSQLContainer<>("postgres:14-alpine");
+        postgres.start();
+
+        String jdbcUrl = postgres.getJdbcUrl();
+        String username = postgres.getUsername();
+        String password = postgres.getPassword();
+
+        // Create the specific database
+        try (var connection = postgres.createConnection("")) {
+            try (var statement = connection.createStatement()) {
+                statement.execute("CREATE DATABASE " + specificDb);
+            }
+        }
+        catch (Exception e) {
+            throw new RuntimeException("Failed to create test database: " + specificDb, e);
+        }
+
+        // Seed data in the main database
+        HaGatewayTestUtils.seedRequiredData(postgres);
+
+        // Also seed data in the specific database
+        String specificDbJdbcUrl = jdbcUrl.replaceAll("/[^/]*$", "/" + specificDb);
+        Jdbi specificJdbi = Jdbi.create(specificDbJdbcUrl, username, password);
+        try (var handle = specificJdbi.open()) {
+            handle.createUpdate(HaGatewayTestUtils.getResourceFileContent("gateway-ha-persistence-postgres.sql"))
+                    .execute();
```

Comment:
Please try initializing the database within the container. https://hub.docker.com/_/postgres#initialization-scripts